c/tskit/trees.c

```diff
@@ -4371,7 +4371,8 @@ tsk_tree_map_mutations(tsk_tree_t *self, int8_t *genotypes,
  * ======================================================== */
 
 int TSK_WARN_UNUSED
-tsk_diff_iter_init(tsk_diff_iter_t *self, tsk_treeseq_t *tree_sequence)
+tsk_diff_iter_init(
+    tsk_diff_iter_t *self, tsk_treeseq_t *tree_sequence, bool include_terminal)
```

Comment:
Let's make this a ``tsk_flags_t options`` argument - this is the approach taken elsewhere in the API, as it's more flexible and breaks less code when new options are added.